indra/sources/signor.py

```diff
@@ -240,134 +299,142 @@ def _get_evidence(row):
-        # Stabilization + up
-        elif row.MECHANISM == 'stabilization' and \
-             row.EFFECT in ('up-regulates', 'up-regulates quantity',
-                            'up-regulates quantity by stabilization'):
-            stmts.append(IncreaseAmount(agent_a, agent_b, evidence=evidence))
-        # Destabilization + down
-        elif row.MECHANISM == 'destabilization' and \
-             row.EFFECT in ('down-regulates', 'down-regulates quantity',
-                            'down-regulates quantity by destabilization'):
-            stmts.append(DecreaseAmount(agent_a, agent_b, evidence=evidence))
-        # Chemical activation + up
-        elif row.MECHANISM == 'chemical activation' and \
-             row.EFFECT in ('up-regulates', 'up-regulates activity'):
-            stmts.append(Activation(agent_a, agent_b, evidence=evidence))
-        # Chemical inhibition + down
-        elif row.MECHANISM == 'chemical inhibition' and \
-             row.EFFECT in ('down-regulates', 'down-regulates activity'):
-            stmts.append(Inhibition(agent_a, agent_b, evidence=evidence))
-        # Binding + Form complex
-        elif row.MECHANISM == 'binding' and row.EFFECT == 'form complex':
-            stmts.append(Complex([agent_a, agent_b], evidence=evidence))
-        # The above mechanism/effect combinations should be the only types
-        # giving rise to statements of the same type with same args.
-        # They also can't give rise to any active form statements; therefore
-        # we have gotten all the statements we will get and can return.
-        if stmts:
-            return (stmts, False)
-
-        # If we have a different effect/mechanism combination, we can now make
-        # them separately without risk of redundancy.
-        # Get the effect statement type:
-        effect_stmt_type = _effect_map[row.EFFECT]
-        # Get the mechanism statement type.
-        if row.MECHANISM:
-            mech_stmt_type = _mechanism_map[row.MECHANISM]
-        else:
-            mech_stmt_type = None
-        # (Note that either or both effect/mech stmt types may be None at this
-        # point.)
-        # First, create the effect statement:
-        if effect_stmt_type == Complex:
-            stmts.append(effect_stmt_type([agent_a, agent_b],
-                                          evidence=evidence))
-        elif effect_stmt_type:
-            stmts.append(effect_stmt_type(agent_a, agent_b, evidence=evidence))
-
-        # For modifications, we create the modification statement as well as
-        # the appropriate active form.
         no_mech = False
-        if mech_stmt_type and issubclass(mech_stmt_type, Modification):
-            if not row.RESIDUE:
-                # Modification
-                mod_stmt = mech_stmt_type(agent_a, agent_b, None, None,
-                                          evidence=evidence)
-                stmts.append(mod_stmt)
+
+        for combo in combinations:
```

Comment:
It's not necessary to generate statements for every possible combination of subject/object members. Instead, can just use the single agents with BoundConditions.